src/db/BotDatabase.js

```diff
@@ -135,10 +135,23 @@ class BotDatabase {
 
     saveLocation(id, coordinates) {
         return this._models.Sender.update(
-            { coordinates: JSON.stringify(coordinates) },
+            { coordinates },
             { where: { id } }
         )
     }
+
+    updateCategories(categories) {
+        const model = this._models.Category;
+        return Promise.all(categories.map(async (item) => {
+            const obj = await model.findOne({ where: { name: item.name } });
+            return obj ? obj.update(item) : model.create(item)
```

Comment:
There is `findOrCreate` method in Sequelize, I think it does pretty similar thing, take a look. Usually I combine `findOrCreate` and then do `update`